server/build_event_protocol/build_event_handler/build_event_handler.go

```diff
@@ -765,30 +800,60 @@ func LookupInvocation(env environment.Env, ctx context.Context, iid string) (*in
 		}
 	}
 
+	invocationMu := sync.Mutex{}
```

Comment:
uber nit: to avoid a mutex, the first goroutine could read the ScoreCard, then after `eg.Wait()` tack the scorecard field onto the invocation

Reply:
ack; I think that's a little more finesse than I want to apply here :P I'd prefer to keep it really straightforward (esp since I don't believe contention will be a problem with only the 2 goroutines).